Engine/Results/RegressionResultHandler.cs

```diff
@@ -76,7 +85,13 @@ public override void SetAlgorithm(IAlgorithm algorithm, decimal startingPortfoli
             }
 
             _writer = new StreamWriter(LogFilePath);
```

Comment:
I think we might want to take the lock on `_sync` here to avoid any race condition leaving some log remaining in the `_preInitializeLines`?

Reply:
Absolutely, will update 👍